packages/connect-wallet-modal/src/Terms/Terms.tsx

```diff
@@ -15,7 +15,9 @@ export const Terms: FC<WalletModalConnectTermsProps> = (props) => {
 
   const handleCheckboxChange = (e: ChangeEvent<HTMLInputElement>) => {
     onChange?.(e);
-    onClickTermsAccept?.();
+    if (e.target.checked) {
```

Comment:
In this case, user can click checkbox to accept terms, but uncheck it right away, revoking their agreement, and, probably, legally he wouldn't be counted as agreed with terms, but we will pass them through